mm2src/coins/eth.rs

```diff
@@ -1462,6 +1463,406 @@ impl WatcherOps for EthCoin {
         // 1.Validate if taker fee is old
     }
 
+    fn taker_validates_taker_payment_refund(&self, input: ValidateWatcherSpendInput) -> ValidatePaymentFut<()> {
```

Comment:
There is a lot of duplicated code between this and `taker_validates_maker_payment_spend`, do you think there is a way to reduce code duplication?
You can work on this refactor on next PR.